File: app/src/main/java/com/example/cleanarchitecture_toyproject/data/net/RestApiService.java
package com.example.cleanarchitecture_toyproject.data.net;

import com.example.cleanarchitecture_toyproject.data.entity.GitHubEntity;
import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface RestApiService {

    //rest 통신을 하기 위한 인터페이스
    @GET("/search/users")
    Observable<GitHubEntity> getUserList(@Query("q") String UserName);
}

File: app/src/main/java/com/example/cleanarchitecture_toyproject/data/net/RestApiService.kt
package com.example.cleanarchitecture_toyproject.data.net

import com.example.cleanarchitecture_toyproject.data.entity.GitHubEntity
import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Query

interface RestApiService {

    //rest 통신을 하기 위한 인터페이스
    @GET("/search/users")
    fun getUserList(@Query("q") UserName: String): Observable<GitHubEntity>
}
